lib/chatter_web/live/chat_live/index.ex

```diff
@@ -111,4 +137,72 @@ defmodule ChatterWeb.ChatLive.Index do
       "=" -> Enum.count(message.likes) == likes_count
     end
   end
+
+  defp filter_messages_by_option(option) do
+    case option do
+      :with_likes_who_liked -> filter_with_likes_who_like()
+      :without_likes_who_never_liked -> filter_without_likes_who_never_liked()
+      :with_major_likes -> filter_with_major_likes()
+    end
+  end
+
+  defp filter_with_likes_who_like do
+    Enum.filter(MessageAgent.get(), fn message ->
+      if Enum.count(message.likes) > 0 && message.author in MessageAgent.get_all_likes(),
+        do: message
+    end)
+  end
+
+  defp filter_without_likes_who_never_liked do
+    Enum.filter(MessageAgent.get(), fn message ->
+      if Enum.empty?(message.likes) && message.author not in MessageAgent.get_all_likes(),
+        do: message
+    end)
+  end
+
+  defp filter_with_major_likes do
+    top_messages =
+      {MessageAgent.get(), Enum.count(MessageAgent.get_all_likes())}
+      |> messages_with_likes_percent()
+      |> Enum.sort_by(&Map.fetch(&1, :likes_percent), :desc)
+      |> top_liked_messages()
+
+    Enum.filter(MessageAgent.get(), fn message -> message.uuid in top_messages.uuids end)
+  end
+
+  defp top_liked_messages(messages) do
+    messages
+    |> Enum.reduce_while(%{uuids: [], percent: 0}, fn message_info, acc ->
+      acc = %{
+        acc
+        | uuids: acc.uuids ++ [message_info.uuid],
+          percent: acc.percent + message_info.likes_percent
+      }
+
+      if acc.percent < 80.0, do: {:cont, acc}, else: {:halt, acc}
+    end)
+  end
+
+  defp messages_with_likes_percent({messages, likes_summary}) do
+    messages
+    |> Enum.reduce([], fn message, acc ->
+      acc ++
+        [
```

Comment:
reduce( ...
[%{} | acc]
...)
|> Enum.reverse()

Reply:
knows now about performance issue with appending to list with list ++ value, fixed